delivery_internal/models/stock_picking.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.depends('carrier_id')
+    def depends_carrier_id(self):
+        print 'Called!'
+        carrier = self.carrier_id
+        if carrier.delivery_type == 'internal':
+            self.env['stock.pickup.request'].create({
```

Comment:
Creating a record inside of a depends is very weird. Won't this logic create a bunch of extra records if I decide to switch the carrier up a few times?

Maybe it would make more sense to maybe hook into one of the [action methods](https://github.com/odoo/odoo/blob/10.0/addons/stock/models/stock_picking.py#L451)?